docs/DeveloperGuide.md

```diff
@@ -1,38 +1,177 @@
+attributes of all other commands within the `commands` package.
 
-* *glossary item* - Definition
+The list command is implemented as follows:
+1. New `ListCommand` object created will call the method `execute` of `Command` class.
+2. The method `execute` decides what to append to the string `printToUser` according to `foodList`. 
+3. `foodList` is an object of type `FoodList` that is passed as argument to `execute`, it consists of the list of food 
+that has been added by user.
+4. `foodList` contains of `Food` that has undergoes the `toString` method.
+5. `execute` will check if `numberOfFood` is greater than 0, in which `numberOfFood` is obtained from the `FoodList` 
+function `getNumberOfFood`.
+6. If `numberOfFood` is greater than 0,  the foodList in String will be appended to `printToUser`. If it is not, then 
+the empty `foodList` will not be appended to `printToUser`.
+7. After appending the `foodList`, the method `execute` will return an object called `CommandResult` and pass          
+`printToUser` as its argument.
+8. `Duke` will then call `printResult` method from `CommandResult` which will print the food list for the user.
+
+**Class Diagram**
+![ClassDiagram](images/ListCommandClassDiagram.png)
+
+**Object Diagram**
+![ClassDiagram](images/ListCommandSequenceDiagram.png)
+
+### Help feature
+The help command is implemented using a `HelpCommand` class which utilizes `COMMAND_WORD` 
+attributes of all other commands within the `commands` package. 
+
+The help command is implemented as follows:
+1. New `HelpCommand` object is created by passing in a String containing arguments from `Parser`.
+2. The constructor `HelpCommand` will split the arguments based on the `--` regex and store them in an array of
+strings called `filters`.
+3. The method `execute` of `Command` class will then be called all the way from `Duke` with `filters` as its argument. 
+4. This method decides what to append to the string `printToUser` as specified by the `filters`. 
+5. After looping through all the `filters`, this method will return an object called `CommandResult` and pass
+`printToUser` as its argument.
+6. `Duke` will then call `printResult` method from `CommandResult` which will print the necessary message.
+
+**Class Diagram**
+![ClassDiagram](images/HelpCommandClassDiagram.png)
+
+
+**Sequence Diagram**
+![ClassDiagram](images/HelpCommandSequenceDiagram.png)
+
+---
+## Appendix: Requirements
+# Product scope
+**Project Direction**
+- Food Supplies Tracker: Main function is to track expiry dates 
+and storage area of foods as to minimise food shortage.
+
+- **Target user profile:**
+- Kitchen worker, Homemaker
+- has a need to manage food
+- prefer desktop apps over other types
+- can type fast
+- prefers typing to mouse interactions
+- is reasonably comfortable using CLI apps
+
+**Value proposition:**
+1) Manage food supply faster than a typical mouse / GUI driven app
```

Comment:
Is the format here planned? Possible suggestion to use 1.,  2. instead so that it appears neater in the DG.